lib/model/classycat_classify.py

```diff
@@ -121,9 +121,18 @@ def classify_and_store_results(self, schema_id, items):
             logger.info(f"Classification results: {classification_results}")
             raise Exception(f"Not all items were classified successfully: "
                             f"input length {len(items)}, output length {len(classification_results)}")
-        # TODO: validate response label against schema https://meedan.atlassian.net/browse/CV2-4801
+
         final_results = [{'id': items[i]['id'], 'text': items[i]['text'], 'labels': classification_results[i]}
                          for i in range(len(items))]
+
+        # filtering out the results that have out-of-schema labels
+        permitted_labels = [topic['topic'] for topic in schema['topics']] + ['Other', 'Unsure']
+        final_results = [result for result in final_results if all(label in permitted_labels for label in result['labels'])]
```

Comment:
maybe helpful to log a warning here for the ones that got dropped?

Reply:
we decided to not drop any of the items, and instead return an empty list for labels, for items that are not labeled properly. it would be good to track percentage of failed requests/classifications but I would count that as the metrics collection work.

Reply:
I agree returning empty list right approach for now, but I think it is important to not do it silently, so I think just logging a warning each time should be easy and sufficient.  Maybe it is rare enough we won't even need metric